packages/cbioportal-utils/src/mutation/MutationAnnotator.ts

```diff
@@ -127,43 +128,39 @@ export function getMutationToTranscriptId(
             mutation,
             variantAnnotation.annotation_summary,
             transcriptConsequenceSummary,
-            true
+            overwriteWithAnnotatedMutation !== undefined
+                ? overwriteWithAnnotatedMutation
+                : true
         );
-        // ignore mutations that don't have a protein change (at some point we
-        // might want to change this to include silent mutations)
-        if (
-            annotatedMutation.proteinChange &&
-            annotatedMutation.proteinChange.length > 0 &&
-            new RegExp(/.*[A-Z].*/, 'i').test(
-                annotatedMutation.proteinChange.toLowerCase()
-            )
-        ) {
-            return annotatedMutation as Mutation;
-        } else {
-            return undefined;
+        // do not ignore mutations that don't have a protein change
+        // include silent mutations
+        if (!annotatedMutation.proteinChange) {
+            annotatedMutation.proteinChange = '';
         }
+        return annotatedMutation as Mutation;
     } else {
-        return undefined;
+        // if mutation is not annotatable or can't map back (due to genomic location normalization or other reasons), return original mutation
+        return mutation;
     }
 }
 
 export function getMutationFromSummary(
```

Comment:
@onursumer @leexgh this function appears to be adding annotation data to the passed mutation.  i think the name of the function should communicate that.   @leexgh i realize this is existing code, but lets improve